src/main/java/org/folio/services/finance/transaction/PaymentCreditWorkflowService.java

```diff
@@ -72,26 +69,23 @@ private List<InvoiceWorkflowDataHolder> withNewPaymentsCredits(List<InvoiceWorkf
   }
 
   private Future<Void> createTransactions(List<InvoiceWorkflowDataHolder> holders, RequestContext requestContext) {
-    Future<Void> future = succeededFuture(null);
-    for (InvoiceWorkflowDataHolder holder : holders) {
-      Transaction tr = holder.getNewTransaction();
-      future = future.compose(v -> baseTransactionService.createTransaction(tr, requestContext)
-        .recover(t -> {
-          if (t instanceof HttpException) {
-            HttpException exception = (HttpException) t;
-            return Future.failedFuture(new HttpException(exception.getCode(), exception.getErrors()));
-          }
-          logger.error("Failed to create transaction for invoice with id - {}", tr.getSourceInvoiceId(), t);
-          List<Parameter> parameters = new ArrayList<>();
-          parameters.add(new Parameter().withKey("invoiceLineId").withValue(tr.getSourceInvoiceLineId()));
-          parameters.add(new Parameter().withKey(FUND_ID)
-            .withValue((tr.getTransactionType() == Transaction.TransactionType.PAYMENT) ? tr.getFromFundId() : tr.getToFundId()));
-          return Future.failedFuture(new HttpException(500, TRANSACTION_CREATION_FAILURE.toError().withParameters(parameters)));
-        })
-        .mapEmpty()
-      );
+    List<Transaction> transactionsToCreate = holders.stream()
+      .map(InvoiceWorkflowDataHolder::getNewTransaction)
+      .toList();
+    if (transactionsToCreate.isEmpty()) {
+      return succeededFuture();
     }
-    return future;
+    return baseTransactionService.batchCreate(transactionsToCreate, requestContext)
+      .recover(t -> {
+        if (t instanceof HttpException he) {
+          return Future.failedFuture(new HttpException(he.getCode(), he.getErrors()));
+        }
+        String invoiceId = transactionsToCreate.get(0).getSourceInvoiceId();
+        logger.error("Failed to create transactions for invoice with id - {}", invoiceId, t);
+        return Future.failedFuture(new HttpException(500, TRANSACTION_CREATION_FAILURE.toError().withParameters(
+          List.of(new Parameter().withKey("invoiceId").withValue(invoiceId))
```

Comment:
now we deleted fundId from list of params, but I checked that it is not used in UI, so I think we are safe to delete this param: https://github.com/folio-org/ui-invoice/blob/master/src/invoices/InvoiceDetails/utils.js#L47 
CC: @usavkov-epam